src/view/treeNodes/treeNode.ts

```diff
@@ -17,6 +18,7 @@ export type TreeNodeParent = TreeNode | BaseTreeNode;
 export const EXPANDED_QUERIES_STATE = 'expandedQueries';
 
 export abstract class TreeNode implements vscode.Disposable {
+	children: TreeNode[];
```

Comment:
It's a little confusing having `children`, `getChildren()` and `cachedChildren()`. Should `children` be private or protected?

Reply:
Yes, `children` should be protected.